pkg/compact/compact.go

```diff
@@ -724,40 +724,49 @@ func (cg *Group) compact(ctx context.Context, dir string, planner Planner, comp
 
 	// Once we have a plan we need to download the actual data.
 	begin := time.Now()
-
 	toCompactDirs := make([]string, 0, len(toCompact))
+
+	var eg errgroup.Group
 	for _, meta := range toCompact {
+		meta := meta
 		bdir := filepath.Join(dir, meta.ULID.String())
+		toCompactDirs = append(toCompactDirs, bdir)
```

Comment:
This is nice, I wonder only, do we always have enough resources (network, mem, CPU) for it to not fail / be slower overall?

Reply:
Now it spawns number of blocks gouroutines. We can use workers to limit the concurrency if the resource is a concern.